f3/granite.go

```diff
@@ -76,6 +77,24 @@ type GMessage struct {
 	Ticket Ticket
 	// Signature by the sender's public key over Instance || Round || Step || Value.
 	Signature []byte
+
+	Justification Justification
+}
+
+// Aggregated list of GossiPBFT messages with the same instance, round and value. Used as evidence for justification of messages
+type Justification struct {
```

Comment:
Justification, in general, has to be exactly the same as the signed message it is based on. This means that the GMessage struct is:
```go
type GMessage struct {
     Sender ActorID
     Current SignedMessage
     CurrentSig Signature
     Ticket Ticket
     Justification SignedMessage
     JustificationSignature Signature
}
```

where SignedMessage contains

Reply:
Makes sense. I instead made `Justification` have two fields `JustificationSignature JustificationSignature` and `Payload SignedMessage`, so that it is easy to carry both around (many times they go together).